File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> map = new HashMap<>();
        for(String[] s : clothes) {
            if(map.containsKey(s[1]))
                map.put(s[1], map.get(s[1])+1);
            else
                map.put(s[1], 1);
        }
        
        Set<Map.Entry<String, Integer>> set = map.entrySet();
        for(Map.Entry<String, Integer> me : set) {
            answer *= (me.getValue()+1);
        }


        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.kt
class Solution {
    fun solution(clothes: Array<Array<String>>): Int {
        val map = mutableMapOf<String, Int>()
        
        for (cloth in clothes) {
            val type = cloth[1]
            map[type] = map.getOrDefault(type, 0) + 1
        }
        
        var answer = 1
        
        for (count in map.values) {
            answer *= count + 1
        }
        
        return answer - 1
    }
}